lib/plugins/aws/package/compile/events/httpApi/index.js

```diff
@@ -130,7 +130,11 @@ class HttpApiEvents {
       ApiId: { Ref: this.provider.naming.getHttpApiLogicalId() },
       StageName: '$default',
       AutoDeploy: true,
+      DefaultRouteSettings: {
```

Comment:
Let's create this configuration only if `provider.httpApi.metrics` is provided (technically `provider.httpApi.metrics != null`)

Reply:
Hey @medikoo , that's what I was initially going for but when testing I noticed that if you've set DefaultRouteSettings.DetailedMetricsEnabled to true, the only way to revert the change is to explicitely set it to false. If you just remove the  DefaultRouteSettings in the CF template it will not deactivate detailed metrics.

(1) This activates metrics
```
"HttpApiStage": {
      "Type": "AWS::ApiGatewayV2::Stage",
      "Properties": {
        "ApiId": {
          "Ref": "HttpApi"
        },
        "StageName": "$default",
        "AutoDeploy": true,
        "DefaultRouteSettings": {
           "DetailedMetricsEnabled": true
        }
}
```

(2) This doesnt deactivate metrics if you've deployed (1) before
```
"HttpApiStage": {
      "Type": "AWS::ApiGatewayV2::Stage",
      "Properties": {
        "ApiId": {
          "Ref": "HttpApi"
        },
        "StageName": "$default",
        "AutoDeploy": true,
}
```

(3) This deactivates metrics if they (1) was deployed previously, if it wasn't it is equivalent to (2)
```
"HttpApiStage": {
      "Type": "AWS::ApiGatewayV2::Stage",
      "Properties": {
        "ApiId": {
          "Ref": "HttpApi"
        },
        "StageName": "$default",
        "AutoDeploy": true,
        "DefaultRouteSettings": {
           "DetailedMetricsEnabled": true
        }
}
```

So if I make it a no setting, users will have to use the console to deactivate detailed metrics.

Reply:
@BaptistG thanks for clarification. Indeed in such case it makes big sense to always configure `DefaultRouteSettings`. Still let's comment in that reasoning (so it's not accidentally removed with some cleanup)